util/chplenv/third-party-pkgs

```diff
@@ -1,92 +1,107 @@
-    if os.access(deffile, os.F_OK) and os.access(deffile, os.R_OK):
-        f = open(deffile, 'r')
-        for line in f.readlines():
-            varvalue = line.strip().split('=')
-            if varvalue[0] == aliasvar:
-                aliases = varvalue[1].split()
-                for a in aliases:
-                    pkg_aliases[a] = dir
-                break
-        f.close()
-
-printchplenv = chpl_home+'/util/printchplenv'
-chplenv=subprocess.Popen([printchplenv, '--simple', '--internal', '--runtime', '--anonymize', '--no-tidy'],
-                         stdout=subprocess.PIPE).communicate()[0].decode()
-
-pkgs=set()
-for l in chplenv.split('\n'):
-    # Consider only lines to that start with 'CHPL_'.
-    if l.startswith('CHPL_'):
-        (var, val) = l.split('=', 1)
-        var=var.strip()
-        val=val.strip()
-        # Ignore platform variables
-        # Ignore CHPL_MEM (use CHPL_JEMALLOC)
-        if not (var.endswith('_PLATFORM') or var == 'CHPL_MEM'):
-            if third_party_pkgs.count(val)==1:
-                pkgs.add(val)
-            elif val in pkg_aliases:
-                pkgs.add(pkg_aliases[val])
-
-# Filter dependencies into 2 list: compiler and runtime dependencies.
-comppkgs=set()
-runpkgs=set()
-for p in pkgs:
-  if os.path.isfile(third_party + "/" + p + "/COMPILERONLY"):
-    comppkgs.add(p)
-  else:
-    runpkgs.add(p)
-
-if len(sys.argv) > 1 and sys.argv[1] == "--compiler":
-  pkgs = comppkgs
-else:
-  pkgs = runpkgs
-
-pkgs_text = ' '.join(pkgs)
-
-print(pkgs_text)
 
+"""
+ Print the names of any third-party packages needed for the current
+ Chapel environment settings
+
+ ASSUMPTIONS:
+ - This script is located in .../chapel/util/chplenv/third-party-pkgs
+ - The format of the output from 'printchplenv' follows the current
+   conventions.  Specifically,
+   - Chapel-specific environment variables are prefixed with 'CHPL_' (***)
+   - Any Chapel environment variable ending in '_PLATFORM' describes
+     the hardware/OS combo and will not considered in this script
+   - "Top-level" Chapel environment variables are printed starting
+     in the first column.  Any Chapel environment variables that 
+     indented are assumed to be modifiers of the corresponding top-level
+     variable and will not be considered in this script (*** and also
+     do not necessarily have to be prefixed with 'CHPL_')
+ - $CHPL_HOME/third-party/<pkg>/Makefile.include defines any aliases
+   for the package name in a variable called XXX_ALIASES where XXX
+   is the directory name in all caps.
+
+ - if the file COMPILERONLY exists in the third party directory,
+   that package is emitted if --compiler is passed,
+   and not by default (which gives dependencies for runtime).
+"""
+
+import optparse
+import os
+import sys
+
+chplenv_dir = os.path.dirname(__file__)
+sys.path.insert(0, os.path.abspath(chplenv_dir))
+
+import utils
+from chpl_home_utils import get_chpl_third_party, get_chpl_home
+
+def get_all_pkgs():
```

Comment:
micro-optimizations (therefore, feel free to ignore):

- Making this a set would give O(1) lookup times (assuming no duplicates and order is not important).
- Memoizing this or storing it into a global variable would also reduce unnecessary `os.listdir()` calls.

Reply:
This is only called a single time from the runtime Makefile (i.e only called during `make`) not during regular chplenv/printchplenv invocations so I'm not worried about micro-optimizations here.

Reply:
I did not realize that. Sounds good.